src/dotnet/commands/dotnet-add/dotnet-add-p2p/Program.cs

```diff
@@ -13,114 +14,101 @@
-                                    Reporter.Error.Write(GetProjectNotCompatibleWithFrameworksDisplayString(
-                                            @ref,
-                                            msbuildProj.GetTargetFrameworks().Select((fx) => fx.GetShortFolderName())));
-                                    return 1;
-                                }
-                            }
-                        }
-                    }
-                    else
+            if (frameworkString == null)
+            {
+                foreach (var tfm in msbuildProj.GetTargetFrameworks())
+                {
+                    foreach (var @ref in refs)
                     {
-                        var framework = NuGetFramework.Parse(frameworkString);
-                        if (!msbuildProj.IsTargettingFramework(framework))
+                        if (!@ref.CanWorkOnFramework(tfm))
                         {
-                            Reporter.Error.WriteLine(string.Format(CommonLocalizableStrings.ProjectDoesNotTargetFramework, msbuildProj.ProjectRootElement.FullPath, frameworkString));
-                            return 1;
-                        }
-
-                        foreach (var @ref in refs)
-                        {
-                            if (!@ref.CanWorkOnFramework(framework))
-                            {
-                                Reporter.Error.Write(GetProjectNotCompatibleWithFrameworksDisplayString(
+                            Reporter.Error.Write(GetProjectNotCompatibleWithFrameworksDisplayString(
                                     @ref,
-                                    new string[] { frameworkString }));
-                                return 1;
-                            }
+                                    msbuildProj.GetTargetFrameworks().Select((fx) => fx.GetShortFolderName())));
+                            return 1;
                         }
                     }
-
-                    var relativePathReferences = references.Select((r) =>
-                        PathUtility.GetRelativePath(msbuildProj.ProjectDirectory, Path.GetFullPath(r))).ToList();
-
-                    int numberOfAddedReferences = msbuildProj.AddProjectToProjectReferences(
-                        frameworkOption.Value(),
-                        relativePathReferences);
-
-                    if (numberOfAddedReferences != 0)
-                    {
-                        msbuildProj.ProjectRootElement.Save();
-                    }
-
-                    return 0;
                 }
-                catch (GracefulException e)
+            }
+            else
+            {
+                var framework = NuGetFramework.Parse(frameworkString);
+                if (!msbuildProj.IsTargettingFramework(framework))
                 {
-                    Reporter.Error.WriteLine(e.Message.Red());
-                    app.ShowHelp();
+                    Reporter.Error.WriteLine(string.Format(
```

Comment:
just wanna confirm - are we still gonna get help for `dotnet add p2p` here? I'm assuming this is handled by command line parser now